src/methods/response.js

```diff
@@ -40,7 +40,9 @@ const parseJsonResponse = function (response, textResponse, responseType) {
 
 const getFetchError = function (error, url, opts) {
   const data = omit(opts, ['Authorization'])
-  Object.assign(error, { name: 'FetchError', url, data })
+  error.name = 'FetchError'
+  error.url = url
+  error.data = data
```

Comment:
We allow direct mutations on variables that are known to be mutable by design. For example, errors should be directly mutated to keep the `Error` prototype. We allow those variables by variable names: https://github.com/netlify/eslint-config-node/blob/d76bd7afca8611d0a00277bdd24435adfdeca38b/.eslintrc.js#L181

However, the `exceptions` option is not available for this specific rule (about `Object.assign()`), so we need to use direct property mutations instead to fix linting.